samples/go/csv/read.go

```diff
@@ -157,18 +157,34 @@ func readFieldsFromRow(row []string, headers []string, fieldOrder []int, kindMap
 	return fields
 }
 
+func primaryKeyFromFields(fields types.ValueSlice, fieldOrder, pkIndices []int) (types.ValueSlice, types.Value) {
+	numPrimaryKeys := len(pkIndices)
```

Comment:
Seems like this len() and the d.Chk could be moved outside of the for loop and therefore not called for every line of a csv file.  The length of pkIndices doesn't change so just call it once and pass it in as an arg here.